Library/Homebrew/formula.rb

```diff
@@ -1939,6 +1939,48 @@ def time
     end
   end
 
+  # Replace duplicate files with links to reduce disk space.
+  #
+  # FIXME: Hardlinks are not fully supported so using `hardlink: true` will only
+  # reduce the bottle size or source build but will be duplicated on bottle pour.
```

Comment:
What makes them not fully supported, out of interest?

Reply:
My old incomplete PR #13154 is related as BSD cp duplicates hardlinks. Trying to switch this brings its own set of problems due to bugs in other commands.

An example of bottle pour behavior is `trino` which has a bottle/estimated-unpack size of <1GB but will pour to >2GB.

There also needs to be special handling when crossing filesystem boundaries (the most extreme case being someone running on a USB formatted as (ex)FAT which doesn't support hardlinks). Of the top of my head, some that can handle this are `rsync -H` and GNU `cp --preserve=links`

Reply:
Thanks for context!

I think an opt-in approach like this makes sense. I think ideally handling this with automatic hardlinks in future (given they are supported in default filesystems on both macOS and Linux) and accepting that you'll get duplication for non-hard links seems ideal.